pkg/resource/snapshot/utility.go

```diff
@@ -0,0 +1,151 @@
+package snapshot
+
+import (
+	"errors"
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	condMsgCurrentlyDeleting string = "snapshot currently being deleted."
+)
+
+var (
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		errors.New("Delete is in progress."),
+		ackrequeue.DefaultRequeueAfterDuration,
+	)
+)
+
+// isDeleting returns true if supplied snapshot resource state is 'deleting'
+func isDeleting(r *resource) bool {
+	if r == nil || r.ko.Status.Status == nil {
+		return false
+	}
+	status := *r.ko.Status.Status
+	return status == "deleting"
```

Comment:
Weird, usually you can find these string constants in the Go SDK - https://docs.aws.amazon.com/sdk-for-go/api/service/memorydb/#pkg-constants

But I can't find anything for `"deleting"`. Would you mind making it a constant at the top of this package, though?